src/DataExplorer/utils.js

```diff
@@ -57,14 +81,100 @@ const queryFileMappingData = async (apiFunc, projectId, selectedTableRows, arran
   return manifestJSON;
 };
 
-const FILENAME = 'manifest.json';
-export const downloadManifest = async (apiFunc, projectId, selectedTableRows, arrangerConfig) => {
+export const downloadManifest = async (
+  apiFunc,
+  projectId,
+  selectedTableRows,
+  arrangerConfig,
+  fileName,
+) => {
   const manifestJSON = await queryFileMappingData(
     apiFunc,
     projectId,
     selectedTableRows,
     arrangerConfig,
   );
   const blob = new Blob([JSON.stringify(manifestJSON, null, 2)], { type: 'text/json' });
-  FileSaver.saveAs(blob, FILENAME);
+  FileSaver.saveAs(blob, fileName);
+};
+
+const getColumns = async (apiFunc, projectId, arrangerConfig) => {
+  const body = {
+    query: `query columnsStateQuery
+            {
+              ${arrangerConfig.graphqlField} {
+                columnsState {
+                  state {
+                    type
+                    keyField
+                    defaultSorted {
+                      id
+                      desc
+                    }
+                    columns {
+                      field
+                      show
+                    }
+                  }
+                }
+              }
+            }`,
+  };
+  const response = await apiFunc({
+    endpoint: `/${projectId}/graphql`,
+    body,
+  });
+  const columns = response.data[arrangerConfig.graphqlField].columnsState.state.columns
+    .filter(col => col.show)
+    .map(col => col.field);
+  return columns;
+};
+
+const querySelectedData = async (
```

Comment:
could this function, `downloadData`, and `queryFileMapping` just be a little more descriptive? This file is pretty dense so I think it'd make it more readable.